Makefile

```diff
@@ -60,10 +60,12 @@ build-slim:
 test: run-unit-tests
 
 .PHONY: lint
+lint: diffnoprefix := $(shell sh -c "git config --get --show-origin diff.noprefix | awk '/file:\.git\/config\s+\S+/ {print \$$2}'")
 lint: build-slim build-test
-	# Note: Make sure that you run `git config diff.noprefix false` in this repo
-	# See this issue for more details: https://github.com/golangci/golangci-lint/issues/948
+	# A workaround for https://github.com/golangci/golangci-lint/issues/948.
+	git config --replace-all diff.noprefix false
```

Comment:
Why not just run `git config diff.noprefix` and if it is false, fail with a clear error (like "run XXX due to bug YYY")?

Then, IIUC, we have the best of both worlds:
 * Not modify git config for the user (specially the replace-all flag that we can't properly restore)
 * A user without that properly set will easily notice


I looked at env vars for git [here][link] and it seems there is no easy way to set it, if not that would be the way to proceed (`ENV=asd cmd`) IMHO. But doesn't seem possible with this option (I had a quick look, so I might be missing something).

[link]: https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables

Reply:
With this approach, you can use `diff.noprefix = true` for this repository and still have properly functional linter. Your approach enforces `noprefix = false`.

Reply:
Yes, enforces that only if you want to run the linter locally. And it makes sense, you can't run the linter locally with that setting. IMHO, it's better to not try to hide that and be clear.

Reply:
What do you think @johananl ?

Reply:
Sorry for the late response. Personally I'd avoid changing Git config on the user's behalf (even if it's local to the repo). IMO this falls under the "taking too much responsibility" category. Like @rata suggested above, I'd print a warning or error out if I have to but I wouldn't run `git config`.